core/src/main/java/org/springframework/security/provisioning/InMemoryUserDetailsManager.java

```diff
@@ -154,6 +161,9 @@ public UserDetails loadUserByUsername(String username) throws UsernameNotFoundEx
 		if (user == null) {
 			throw new UsernameNotFoundException(username);
 		}
+		if (user instanceof CredentialsContainer) {
+			return user;
+		}
 		return new User(user.getUsername(), user.getPassword(), user.isEnabled(), user.isAccountNonExpired(),
 				user.isCredentialsNonExpired(), user.isAccountNonLocked(), user.getAuthorities());
 	}
```

Comment:
Can you also please update `updateUser` in the same way?

Reply:
Yes, done.